CDCgov/prime-reportstream
# Remove unused `memberships` Map and pattern

This PR removes the idea of having more than one group membership within our authentication service. All users are expected to be in **one group only**, and for admins, we will default to their admin group over any non-admin group. 

Also in this PR, we refactor our test suite for `useOktaMemberships` to decouple it from the idea of selecting memberships from the removed `memberships` map.

Tangential work to #7923 to simplify the addition of services in a memberhsip.

Test Steps:
1. Ensure you can log in, and still operate within the application
2. Observe your membership state via React Dev Tools (find SessionProvider and view its state) or localStorage to ensure no `memberships` map was created/populated
3. Run the full test suite to ensure no failing tests

## Changes
- Removes `memberships` map 
- Refactors `useOktaMemberships` tests

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
